docker/Dockerfile

```
@@ -0,0 +1,46 @@
+FROM node:10.16.3@sha256:e0f7dabe991810057aee6ba7a7d4136fe7fc70e99235d79e6c7ae0177656a5c8
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+        gettext \
+        mime-support \
+        build-essential \
+        python2.7 \
+        default-jre \
+    && rm -rf /var/lib/apt/lists/*
+
+# remove the node user from the base package and add a non-privileged user
+RUN userdel --force --remove node && \
+    adduser --uid 1000 --disabled-password --gecos '' --no-create-home kumascript
+
+ARG REVISION_HASH
+# make the git commit hash permanently available within this image.
+ENV REVISION_HASH $REVISION_HASH
+
+WORKDIR /
+
+COPY kumascript/package.json kumascript/npm-shrinkwrap.json /
+RUN npm config set python /usr/bin/python2.7 && \
+    # install the Node.js dependencies,
+    # with versions specified in npm-shrinkwrap.json
+    npm ci && \
```

`npm ci` is an alias to `npm clean‑install`
```suggestion
    npm clean-install && \
```

---

@ExE-Boss It's my understanding that `npm ci` is the proper command. In fact, I don't see any `npm` documentation on a `npm clean-install` command, and I think the `ci` references `continuous integration` rather than `clean install`. In any case, I'm going to keep this as it is, since changing it would be outside the scope of this PR anyway.

---

It is as of https://github.com/npm/cli/pull/57.

---

Oh! Good to know, thanks @ExE-Boss, but I'll leave that for another PR.